integrations/integration-config.php

```diff
@@ -0,0 +1,176 @@
+	private array $config = [];
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $slug Slug of the integration.
+	 */
+	public function __construct( string $slug ) {
+		$this->set_config( $slug );
+	}
+
+	/**
+	 * Set config provided by VIP from file.
+	 *
+	 * @param string $slug A unique identifier for the integration.
+	 */
+	private function set_config( string $slug ): void {
+		$config = $this->get_vip_config_from_file( $slug );
+
+		if ( ! is_array( $config ) ) {
+			return;
+		}
+
+		$this->config = $config;
+	}
+
+	/**
+	 * Get config provided by VIP from file.
+	 *
+	 * @param string $slug A unique identifier for the integration.
+	 *
+	 * @return null|mixed
+	 */
+	protected function get_vip_config_from_file( string $slug ) {
+		$config_file_path = ABSPATH . 'config/integrations-config/' . $slug . '-config.php';
+
+		if ( ! is_readable( $config_file_path ) ) {
+			return null;
+		}
+
+		return require_once $config_file_path;
+	}
+
+	/**
+	 * Returns `true` if the integration is enabled in VIP config else `false`.
+	 *
+	 * @return bool
+	 *
+	 * @private
+	 */
+	public function is_active_via_vip(): bool {
+		// Return false if blocked on client.
+		if ( $this->get_value_from_config( 'client', 'status' ) === Client_Integration_Status::BLOCKED ) {
+			return false;
+		}
+
+		$site_status = $this->get_value_from_config( 'site', 'status' );
+
+		// Return false if blocked on site.
+		if ( Site_Integration_Status::BLOCKED === $site_status ) {
+			return false;
+		}
+
+		// Look into network_sites config before because if not present we will fallback to site config.
+		$network_site_status = $this->get_value_from_config( 'network_sites', 'status' );
+
+		if ( Site_Integration_Status::ENABLED === $network_site_status ) {
+			return true;
+		}
+
+		// Return false if status is defined but other than enabled. If status is not defined then fallback to site config.
+		if ( null !== $network_site_status ) {
+			return false;
+		}
+
+		// Return true if enabled on site.
+		return Site_Integration_Status::ENABLED === $site_status;
+	}
+
+	/**
+	 * Get site config.
+	 *
+	 * @return null|array
+	 *
+	 * @private
+	 */
+	public function get_site_config() {
+		if ( is_multisite() ) {
+			return $this->get_value_from_config( 'network_sites', 'config' );
+		}
+
+		return $this->get_value_from_config( 'site', 'config' );
+	}
+
+	/**
+	 * Get config value based on given type and key.
+	 *
+	 * @param string $config_type Type of the config whose data is needed i.e. client, site, network-sites etc.
+	 * @param string $key Key of the config from which we have to extract the data.
+	 *
+	 * @return null|array
+	 *
+	 * @throws InvalidArgumentException Exception if invalid argument is passed.
+	 */
+	protected function get_value_from_config( string $config_type, string $key ) {
+		if ( ! in_array( $config_type, [ 'client', 'site', 'network_sites' ], true ) ) {
+			throw new InvalidArgumentException( 'Config type must be one of client, site and network_sites.' );
```

Comment:
Avoid fatals:

```suggestion
			trigger_error( '$config_type param must be one of client, site and network_sites.', E_USER_WARNING );
			return null;
```

Aside: It may be nice to have a helper function that can throw the exception locally but avoid fatals in production.